core/bot_framework_adapter.go

```diff
@@ -92,12 +93,32 @@ func (bf *BotFrameworkAdapter) ProcessActivity(ctx context.Context, req schema.A
 
 	replyActivity, err := activity.PrepareActivityContext(handler, turnContext)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "Failed to create Activity context.")
+	}
+
+	response, err := activity.NewActivityResponse(bf.Client)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create response object.")
+	}
+
+	return response.SendActivity(replyActivity)
+}
+
+// ProactiveMessage sends activity to a conversation.
+// This methods is used for Bot initiated conversation.
+func (bf *BotFrameworkAdapter) ProactiveMessage(ctx context.Context, ref schema.ConversationReference, handler activity.Handler) error {
+	turnContext := &activity.TurnContext{
+		Activity: activity.ApplyConversationReference(schema.Activity{Type: schema.Message}, ref, true),
+	}
+
+	replyActivity, err := activity.PrepareActivityContext(handler, turnContext)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create Activity context.")
 	}
 
 	response, err := activity.NewActivityResponse(bf.Client)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "Failed to create response object.")
```

Comment:
@PrasadG193 ProcessActivity and ProactiveMessage function have some same code. Can we take that code out?

Reply:
Sounds good!